### PR TITLE
Impl HoaConsole::redisplay()

### PR DESCRIPTION
**Pull request by zonuexe**
suggested by @Hywan https://github.com/bobthecow/psysh/pull/300#discussion_r64150509

### CODE REVIEW

**Comment by Hywan** [comment]
You have to reposition the cursor too. Don't know why it changes its position though.

File: src/Psy/Readline/HoaConsole.php [hunk 1]
```diff
@@ -94,7 +101,9 @@ public function readline($prompt = null)
      */
     public function redisplay()
     {
-        // noop
+        $current_line = $this->hoaReadline->getLine();
+        Cursor::clear('all');
```
Comment (Hywan) [inline src/Psy/Readline/HoaConsole.php]: Maybe add a `Cursor::moveTo(0, 0)` here?

Comment (zonuexe) [inline src/Psy/Readline/HoaConsole.php]: The following code also did not have effect which I imagined.

```php
        $current_line = $this->hoaReadline->getLine();
        Cursor::clear('all');
        //Cursor::moveTo(0, 0);
        echo $this->lastPrompt, $current_line;
        Cursor::moveTo(strlen($this->lastPrompt) + strlen($current_line) + 1, 0);
```

Comment (Hywan) [inline src/Psy/Readline/HoaConsole.php]: You should use `mb_strlen` in case of UTF-8 (assuming the terminal supports UTF-8).
What happened?

**Comment by zonuexe** [comment]
Certainly `mb_strlen()` is appropriate than `strlen()`, but it seems there is no effect on our face problems.

hoa-console-redisplay-termapp

**Comment by Hywan** [comment]
What terminal are you using?

**Comment by zonuexe** [comment]
@Hywan 
I'm using Terminal.app (2.6.1, El Capitan) and iTerm2 (Build 3.0.20160607-nightly).
Both seem to have same behavior.

**Comment by zonuexe** [comment]
@bobthecow @Hywan 
I have forgotten this Pull Request for several years. This feature now works perfectly.

**Comment by bobthecow** [comment]
> I have forgotten this Pull Request for several years. This feature now works perfectly.

Ha! Thanks for the update :)

Development has moved to the `master` branch since then. Do you mind rebasing against that?

**Comment by zonuexe** [comment]
I rebased this branch.

File: src/Readline/HoaConsole.php [hunk 2]
```diff
@@ -94,7 +101,19 @@ public function readline($prompt = null)
...
-        // noop
+        $this->_redisplay();
+    }
+
+    /**
+     * @return int
+     */
+    public function _redisplay()
+    {
+        $current_line = $this->hoaReadline->getLine();
```
Comment (bobthecow) [inline src/Readline/HoaConsole.php]: Do you mind updating this variable name to camelCase to match the convention in the codebase?

Comment (zonuexe) [reply src/Readline/HoaConsole.php]: OK, I fix it.

File: src/Readline/HoaConsole.php [hunk 3]
```diff
@@ -22,6 +23,9 @@ class HoaConsole implements Readline
     /** @var HoaReadline */
     private $hoaReadline;
 
+    /** @var string */
```
Comment (GrahamCampbell) [inline src/Readline/HoaConsole.php]: `string|null`

**Comment by bobthecow** [comment]
Thanks again :)

File: src/Readline/HoaConsole.php (hunk 3 above)
Comment (zonuexe) [reply src/Readline/HoaConsole.php]: Thank you! I will fix it in other PR.

Comment (GrahamCampbell) [reply src/Readline/HoaConsole.php]: Don't worry. I've already sent one.

**Pull request merged**